content/200-orm/200-prisma-client/500-deployment/301-edge/450-deploy-to-cloudflare.mdx

```diff
@@ -0,0 +1,526 @@
+
+- use the `@prisma/adapter-pg` database adapter (via the `driverAdapters` Preview feature)
+- set `node_compat = true` in `wrangler.toml` (see the [Cloudflare docs](https://developers.cloudflare.com/workers/wrangler/configuration/#add-polyfills-using-wrangler))
+
+If you are running into a size issue and can't deploy your application because of that, you can use our slimmer variant of the `pg` driver package [`@prisma/pg-worker`](https://github.com/prisma/prisma/tree/main/packages/pg-worker) and the [`@prisma/adapter-pg-worker`](https://github.com/prisma/prisma/tree/main/packages/adapter-pg-worker) adapter that belongs to it.
+
+`@prisma/pg-worker` is an optimized and lightweight version of pg that is designed to be used in a worker. It is a drop-in replacement for `pg` and is fully compatible with Prisma ORM.
+
+#### 1. Configure Prisma schema & database connection
+
+First, ensure that database connection is configured properly. In your Prisma schema, set the `url` of the `datasource` block to the `DATABASE_URL` environment variable. You also need to enable the `driverAdapters` feature flag:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+Next, you need to set the `DATABASE_URL` environment variable to the value of your database connection string. You'll do this in a file called `.dev.vars` used by Cloudflare:
+
+```bash file=.dev.vars
+DATABASE_URL="postgresql://admin:mypassword42@somehost.aws.com:5432/mydb"
+```
+
+Because the Prisma CLI by default is only compatible with `.env` files, you can adjust your `package.json` with the following script that loads the env vars from `.dev.vars`. You can then use this script to load the env vars before executing a `prisma` command.
+
+Add this script to your `package.json`:
+
+```js file=package.json highlight=5;add
+{
+  // ...
+  "scripts": {
+    // ....
+    "env": "dotenv -e .dev.vars"
+  },
+  // ...
+}
+```
+
+Now you can execute Prisma CLI commands as follows while ensuring that the command has access to the env vars in `.dev.vars`:
+
+```terminal
+npm run env -- npx prisma
+```
+
+#### 2. Install dependencies
+
+Next, install the required packages:
+
+```terminal
+npm install @prisma/adapter-pg
+npm install pg
+npm install @types/pg --save-dev # if you're using TypeScript
+```
+
+#### 3. Set `node_compat = true` in `wrangler.toml`
+
+In your `wranger.toml` file, add the following line:
+
+```toml file=wranger.toml
+node_compat = true
+```
+
+> **Note**: For Cloudflare Pages, using `node_compat` is not officially supported, but you can use a pre-release version of the CLI that adds support for `--node-compat` to `pages deploy`. Learn more in this [feature request](https://github.com/cloudflare/workers-sdk/issues/1074).
```

Comment:
it's less concrete but since the workaround seems complicated I don't know if we want to pull it into the docs directly... I'd rather link to it from a note.